src/input.rs

```diff
@@ -3,26 +3,47 @@
 use crate::{
     action::{GameAction, GameActionQueue},
     camera::{CameraControl, CameraTarget},
-    character::Movement,
-    interface::MenuLayer,
-    map::{MapPosition, MapPresence, PathGuided, Zone},
+    camp::Camp,
+    character::Character,
+    interface::{shell::handle_action_button_interaction, MenuLayer},
+    map::{GameMap, MapPosition, MapPresence, PathGuided, Zone},
+    party::{Group, Party},
+    selection::NextSelectionQuery,
+    State,
 };
 use bevy::prelude::*;
 use bevy_mod_picking::{DefaultPickingPlugins, PickingEvent, Selection};
 use leafwing_input_manager::plugin::InputManagerSystem;
-pub use leafwing_input_manager::prelude::*;
+pub use leafwing_input_manager::{common_conditions::action_just_pressed, prelude::*};
+use smallvec::SmallVec;
 
 pub struct InputPlugin;
 
 impl Plugin for InputPlugin {
     fn build(&self, app: &mut App) {
         app.add_plugins(DefaultPickingPlugins)
             .add_plugin(InputManagerPlugin::<Action>::default())
-            .add_startup_system(spawn_input_manager)
-            .add_systems((handle_deselect, handle_select_next))
+            .init_resource::<ActionState<Action>>()
+            .insert_resource(input_map())
+            .add_systems(
+                (
+                    handle_deselect.run_if(action_just_pressed(Action::Deselect)),
+                    handle_select_next.run_if(action_just_pressed(Action::SelectNext)),
+                    handle_resume_move.run_if(action_just_pressed(Action::ResumeMove)),
+                    handle_camp.run_if(action_just_pressed(Action::Camp)),
+                    handle_break_camp.run_if(action_just_pressed(Action::BreakCamp)),
+                    handle_create_party.run_if(action_just_pressed(Action::CreateParty)),
+                    handle_split_party.run_if(action_just_pressed(Action::SplitParty)),
+                    handle_merge_party.run_if(action_just_pressed(Action::MergeParty)),
+                    handle_collect_crystals.run_if(action_just_pressed(Action::CollectCrystals)),
+                )
+                    .after(handle_action_button_interaction)
```

Comment:
not a fan of how this pokes into what otherwise could be an internal systems of the interface